adminapi/request.py

```diff
@@ -35,22 +35,25 @@ def send_request(endpoint, data, auth_token, timeout=None):
         auth_token = get_auth_token()
 
     data_json = json.dumps(data, default=json_encode_extra)
-    try_backup = False
+    retries = 3
 
-    try:
-        req = _build_request(endpoint, auth_token, data_json, try_backup)
-        return json.loads(urlopen(req, timeout=timeout).read().decode('utf8'))
-    except HTTPError as error:
-        if error.code in (500, 502):
-            try_backup = True
-        else:
-            raise
-    except URLError:
-        try_backup = True
+    while retries > 0:
+        try:
+            req = _build_request(endpoint, auth_token, data_json, False)
+            return json.loads(
+                urlopen(req, timeout=timeout).read().decode('utf8'))
+        except HTTPError as error:
+            if error.code not in (500, 502):
+                raise
+        except URLError:
+            pass
 
-    if try_backup:
-        req = _build_request(endpoint, auth_token, data_json, try_backup)
-        return json.loads(urlopen(req, timeout=timeout).read().decode('utf8'))
+        # In case of an api error, sleep 5 seconds and try again three times
+        time.sleep(5)
+        retries -= 1
+
+    req = _build_request(endpoint, auth_token, data_json, True)
+    return json.loads(urlopen(req, timeout=timeout).read().decode('utf8'))
```

Comment:
It would be nicer to eliminate this duplicate code now.  We can check do `if not retry: raise`.